src/plugins/codeStyle/index.ts

```diff
@@ -317,6 +332,53 @@ export default class CodeStyle {
         }
     }
 
+    validateCreateObject(s: CallExpression, diagnostics: (Omit<BsDiagnostic, 'file'>)[], severity: DiagnosticSeverity) {
+        if (!this.isCreateObject(s)) {
+            return;
+        }
+        const callParams = this.getLiteralArgs(s.args);
+
+        if (callParams && callParams.length === 3 && callParams[0] === 'roRegex') {
+            const parentSt = s.findAncestor((node, cancel) => {
+                if (isIfStatement(node)) {
+                    cancel.cancel();
+                } else if (this.isLoop(node) || isFunctionStatement(node)) {
+                    return true;
+                }
+            });
+
+            let hasIssue = false;
+            if (isFunctionStatement(parentSt)) {
+                for (const callExp of parentSt.func.callExpressions) {
```

Comment:
If I'm reading this logic correctly, I think this results in a significant amount of extra work which gets worse for every additional `createObject` we find.

Consider the following brightscript code:

```
function func1()
    CreateObject("roRegex", "test1", "")
    CreateObject("roRegex", "test2", "")
    CreateObject("roRegex", "test3", "")
end function
```

Here's my understanding of the current implementation:
```typescript
for (const callExpression of allCallsInTheAst) {
    var parentFunction= callExpression.findparentFunction();
    for (const call in parentFunction.callExpressions) {
         'do the work
    }
}
```

Which would result in processing each regexp 3 times (2 excess times)
- found `test1`. found `func1`. loop over `test1`,` test2`,` test3`
- found `test2`. found `func1`.loop over `test1`, `test2,` `test3`
- found `test3`. found `func1`. loop over `test1`, `test2`, `test3`


And here's how I think it can be improved:
```
for (const functionExpression of allFunctionExpressionsInAST) {
    for( const callExpression of parentFunction.callExpressions) {
         'do the work
    }
}
```

Which would result in processing each regexp exactly 1 time
- found `func1`. loop over `test1`, `test2`, `test3`